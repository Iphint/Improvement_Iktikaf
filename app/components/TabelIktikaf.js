import React from 'react';
import { StyleSheet, Text, View, ScrollView, Alert } from 'react-native';

const TabelIktikaf = ({ jumlahDoa }) => {
 const dataMalam = [];
 for (let i = 21; i <= 30; i++) {
    dataMalam.push({
      malam: `Malam ${i}`,
      jumlahDoa: jumlahDoa[i] || 0,
    });
 }

 return (
   <ScrollView style={styles.container} showsVerticalScrollIndicator={false}>
   <View style={styles.header}>
     <Text style={styles.headerText}>Malam</Text>
     <Text style={styles.headerText}>Jumlah Doa</Text>
   </View>
   {dataMalam.map((item, index) => {
     let backgroundColor;
     if (item.jumlahDoa >= 1000) {
       backgroundColor = 'blue';
       Alert.alert(
         'Congratulation !!!',
         `You've passed 1000 times reading du'a, but you should be more spirit tommorow, keep spirit`,
         [
           {
             text: 'OK',
             onPress: () => console.log('OK Pressed'),
           },
         ],
         { cancelable: false }
       );
     } else if (item.jumlahDoa >= 600) {
       backgroundColor = 'green';
     } else if (item.jumlahDoa >= 200) {
       backgroundColor = 'orange';
     } else {
       backgroundColor = 'white';
     }

     return (
       <View key={index} style={[styles.row, { backgroundColor }]}>
         <Text style={styles.cell}>{item.malam}</Text>
         <Text style={styles.cell}>{item.jumlahDoa}</Text>
       </View>
     );
   })}
 </ScrollView>
 );
};

export default TabelIktikaf;

const styles = StyleSheet.create({
 container: {
    flex: 1,
    backgroundColor: '#fff',
 },
 header: {
    flexDirection: 'row',
    backgroundColor: '#f2f2f2',
    paddingVertical: 10,
    paddingHorizontal: 20,
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
 },
 headerText: {
    flex: 1,
    fontWeight: 'bold',
 },
 row: {
    flexDirection: 'row',
    paddingVertical: 10,
    paddingHorizontal: 20,
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
 },
 cell: {
    flex: 1,
 },
});
