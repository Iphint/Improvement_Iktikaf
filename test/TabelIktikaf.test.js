import React from 'react';
import renderer from 'react-test-renderer'
import TabelIktikaf from '../app/components/TabelIktikaf';

const dummyData = {
    21: 150,
    22: 250,
    23: 350,
    24: 450,
    25: 550,
    26: 650,
    27: 750,
    28: 850,
    29: 950,
    30: 1050,
  };

  describe('TabelIktikaf', () => {
    it('renders correctly with dummy data', () => {
       const tree = renderer
         .create(<TabelIktikaf jumlahDoa={dummyData} />)
         .toJSON();
       expect(tree).toMatchSnapshot();
    });
   });