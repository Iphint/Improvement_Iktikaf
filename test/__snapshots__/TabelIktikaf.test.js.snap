// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabelIktikaf renders correctly with dummy data 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        {
          "backgroundColor": "#f2f2f2",
          "borderBottomColor": "#ccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "flex": 1,
            "fontWeight": "bold",
          }
        }
      >
        Malam
      </Text>
      <Text
        style={
          {
            "flex": 1,
            "fontWeight": "bold",
          }
        }
      >
        Jumlah Doa
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          {
            "backgroundColor": "white",
          },
        ]
      }
    >
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        Malam 21
      </Text>
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        150
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          {
            "backgroundColor": "orange",
          },
        ]
      }
    >
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        Malam 22
      </Text>
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        250
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          {
            "backgroundColor": "orange",
          },
        ]
      }
    >
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        Malam 23
      </Text>
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        350
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          {
            "backgroundColor": "orange",
          },
        ]
      }
    >
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        Malam 24
      </Text>
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        450
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          {
            "backgroundColor": "orange",
          },
        ]
      }
    >
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        Malam 25
      </Text>
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        550
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          {
            "backgroundColor": "green",
          },
        ]
      }
    >
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        Malam 26
      </Text>
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        650
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          {
            "backgroundColor": "green",
          },
        ]
      }
    >
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        Malam 27
      </Text>
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        750
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          {
            "backgroundColor": "green",
          },
        ]
      }
    >
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        Malam 28
      </Text>
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        850
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          {
            "backgroundColor": "green",
          },
        ]
      }
    >
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        Malam 29
      </Text>
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        950
      </Text>
    </View>
    <View
      style={
        [
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          {
            "backgroundColor": "blue",
          },
        ]
      }
    >
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        Malam 30
      </Text>
      <Text
        style={
          {
            "flex": 1,
          }
        }
      >
        1050
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
